### PR TITLE
File handling for kubeconfig

### PR DESCRIPTION
**Pull request by TerryHowe**
I'd eventually like to get away from creating the files, but the way helm works this is a lot easier.

### CODE REVIEW

File: pkg/authenticator/kubeconfig_test.go [hunk 1]
```diff
@@ -35,6 +35,7 @@ func TestManagerContext_getRegistry(t *testing.T) {
 		fileName, err := sut.GetKubeconfig(ctx, "billy")
 		assert.NoError(t, err)
 		assert.Equal(t, "billy-kubeconfig", fileName)
-		os.Remove(fileName)
+		err = os.Remove(fileName)
+		assert.NoError(t, err)
```
Comment (ivyostosh) [inline pkg/authenticator/kubeconfig_test.go:39]: Should we use `require.NoError` more consistently across different tests?

Comment (TerryHowe) [reply pkg/authenticator/kubeconfig_test.go:39]: This is just informative on the clean-up. The require.NoError stops the tests which would be fine here, but this error is not very important

File: pkg/authenticator/kubeconfig_test.go [hunk 2]
```diff
@@ -35,6 +35,7 @@ func TestManagerContext_getRegistry(t *testing.T) {
 		fileName, err := sut.GetKubeconfig(ctx, "billy")
 		assert.NoError(t, err)
```
Comment (ewollesen) [inline pkg/authenticator/kubeconfig_test.go:36]: (This one could probably be a require.NoError though 😁)

File: pkg/authenticator/kubeconfig.go [hunk 3]
```diff
@@ -43,6 +43,11 @@ func (kc *kubeconfigClient) GetKubeconfig(ctx context.Context, clusterName strin
 		return "", fmt.Errorf("getting kubeconfig for cluster %q: %w", clusterName, err)
 	}
 
+	if _, err := os.Stat(secretName); err == nil {
+		_ = os.Chown(secretName, os.Geteuid(), os.Getegid())
```
Comment (ewollesen) [inline pkg/authenticator/kubeconfig.go:47]: I would have returned these as errors. Better to know sooner than later that there's a problem.

I'm guessing you disagree, but would you be amenable to at least a log message, so we'd have some clue that there's something wrong?

Comment (TerryHowe) [reply pkg/authenticator/kubeconfig.go:47]: 99% sure the error would be permission denied which will show up below on the write. I don't want to clutter this up with code I think is a bit throw away away. Hopefully, we move away from this file approach.

**Pull request merged**